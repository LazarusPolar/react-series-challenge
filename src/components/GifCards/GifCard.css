.GifCards {
    z-index: 10px;
    display: inline-block;
}

.GifCards img {
    width: 400px;
    height: 250px;
    z-index: 10px;
    display: inline-block;
}