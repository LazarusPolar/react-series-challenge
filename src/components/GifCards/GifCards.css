.GifCards {
    z-index: 10px;
    display: inline-block;
}

.GifCards img {
    width: 250px;
    height: 150px;
    z-index: 10px;
    display: inline-block;
}