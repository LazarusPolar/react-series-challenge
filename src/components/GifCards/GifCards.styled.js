import styled from 'styled-components';

// Super awesome box shadow effect provided from W3Schools
// Awesome job link: https://www.w3schools.com/cssref/tryit.asp?filename=trycss3_box-shadow4
export const GifCardsStyled = styled.div`
    display: inline-block;
    background-color: ${props => props.theme.favoritePageBackground};
    padding: 10px;
    margin: 10px;
    border: 1px solid;
    padding: 10px;
    box-shadow: 5px 5px 8px blue, 10px 10px 8px red, 15px 15px 8px green;
`

export const StyledH1 = styled.h1`
    color: #ffffff;
    font-family: 'Quicksand', sans-serif;  
`