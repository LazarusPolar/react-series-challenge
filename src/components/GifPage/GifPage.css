.GifPage {
    z-index: 10px;
    display: inline-block;
}

.GifPage GifCards {
    width: 400px;
    height: 250px;
    z-index: 10px;
    display: inline-block;
}