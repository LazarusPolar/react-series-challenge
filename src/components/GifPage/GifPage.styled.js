import styled from 'styled-components';

export const StyledGifPage = styled.div`
  text-align: center;
  padding-left: 0px;
  padding-right: 2px;
  width: 100%;
  background-color: DarkSlateGray;
`
