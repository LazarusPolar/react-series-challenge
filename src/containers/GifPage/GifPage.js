import React, {Component} from 'react'

class GifPage extends Component {
    constructor(props){
        super(props);
        this.state = {
            variable: 1
        };
    }

    render() {
        return(
        <div>
            <header>
                <nav>
                    <ul>
                        <li>Home</li>
                        <li>Favorites</li>
                    </ul>
                </nav>
            </header>

            <div>
                <input type="text" placeholder = "Search for awesome gifs" />
            </div>
        </div>)
    }
}

export default GifPage;