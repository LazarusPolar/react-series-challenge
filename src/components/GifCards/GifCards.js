import React from 'react';
import GifCard from '../GifCards/GifCard/GifCard'
import GifCardControls from '../GifCards/GifCardControls/GifCardControls'
import './GifCards.css';

const GifCards = (props) => {

        let arrays = null;
        const { gifData, searchedValue, searchedGifs } = props; 
        //console.log("props", props); 

        if(gifData !== null && gifData !== undefined){
            arrays = Object.keys(gifData).map((key, index) => {
                let { id, url } = gifData[index];

                console.log("url: " + url);
                //return (<img key = {id} src = {url} alt = "Cargando..." />);
                return (
                <div key = {id} className = "GifCards">
                <GifCard 
                    //loadError = {props.loadError}
                    //loadSuccess = {props.loadSuccess}
                    imageUrl = {url}
                     />
                    <GifCardControls key = {index} searchedGifs = {() => searchedGifs({ id: id, searchedValue: searchedValue, url: url})} />
                </div>);
            });

            //console.log("Arreglo" , arrays[0]);
            if(arrays.length === 0){
                arrays = <h1> No results :( </h1>;
            }
        }

        /*
        if(this.state.isLoading) {
            image = <p> Cargando imagen... </p>
        } else {
            //TODO: Spinner
            loading = <p> Cargando </p>
            image = <img src = {this.props.imageUrl} onLoad = {this.handleOnLoad} onError = {this.handleError} />
        }*/

        return(
            <div className = "GifCards">
                { arrays }
            </div>
        )
    
}

export default GifCards;