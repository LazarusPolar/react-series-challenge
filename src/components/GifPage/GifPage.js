import React, {Component} from 'react';
import GifCards from '../GifCards/GifCards'
import './GifPage.css';
import SearchBar from '../../UI/SearchBar';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import SearchFunctions from '../../actions/searchValues';
import ErrorBoundary from '../../ErrorBoundary/ErrorBoundary';

class GifPage extends Component {

    componentDidMount() {      
        this.props.requestApiData();
    }

    handleSearch = () => {
        this.props.requestApiDataSearched();
    }

    updateState = (event) => {
        this.props.updateSearchValue({searchValue: event});
    }

    render() {
        let newArray = [];
        let newObject = {};

        if(this.props.gifs !== undefined ){
            for(let a = 0; a <= Object.keys(this.props.gifs).length - 1; a++){
                newObject = {};
                newObject.id = this.props.gifs[a].id;
                newObject.url = this.props.gifs[a].images.downsized.url;
                newArray.push(newObject);
            }
        }

        return(
            <div className = "GifPage">
                <SearchBar updateState = { this.updateState } handleSearch = {this.handleSearch} />
                <ErrorBoundary>
                <GifCards gifData = { newArray } addRemoveGif = {this.props.addRemoveGifFavorites} searchedValue = {this.props.searchedValue}/> 
                </ErrorBoundary>        
            </div>)
    }
}

const mapStateToProps = (state) => {
    const { gifs, searchedValue } = state;
    return {
        gifs, searchedValue
    }
}

const mapDispatchToProps = (dispatch) => {
    const { addRemoveGifFavorites, updateSearchValue, requestApiData, requestApiDataSearched } = SearchFunctions.creators;
    return bindActionCreators( { addRemoveGifFavorites, updateSearchValue, requestApiData, requestApiDataSearched }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(GifPage);