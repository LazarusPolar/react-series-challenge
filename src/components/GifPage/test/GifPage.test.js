import GifCards from '../../GifCards/GifCards';
import ErrorBoundary from '../../../ErrorBoundary/ErrorBoundary'
import { StyledGifPage } from '../GifPage.styled'
import SearchBar from '../../../UI/SearchBar'
import React from 'react';
import { shallow } from 'enzyme';
import GifPage from '../GifPage';

describe('GifFavorite', () => {
  const func = jest.fn();
  const searchResult = [{
      id: 1,
      searchedValue: "trendy",
      url: "https://image", 
      isFavorite: true,
  }]
    const searchResultEmpty = [];
    const themeSunny = {
        gifPageBackground: '#969ec3',
        favoritePageBackground: '#c8bed7',
        headerBackground: '#fde1cc',
        cardBackground: '#fec492',
    }
    let wrapper;
/*
    beforeEach(() => {
        wrapper = shallow(
            <GifPage />
        );
    });

    test('should render', () => {
        expect(wrapper).toMatchSnapshot();
    });

    
    test('should set GifCards ', () => {
        expect(wrapper.contains(
            <StyledGifPage {...themeSunny}>
                <SearchBar 
                updateState = { func } 
                handleSearch = { func } />

                <ErrorBoundary>
                    <GifCards 
                    gifData = { searchResult } 
                    addRemoveGif = { func } 
                    searchedValue = { func }/> 
                </ErrorBoundary>        
            </StyledGifPage>
        )).toBe(true);
    });*/

})
