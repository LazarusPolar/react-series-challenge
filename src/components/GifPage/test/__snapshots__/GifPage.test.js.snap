// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifFavorite should render 1`] = `undefined`;
